test/e2e/deployment.go

```diff
@@ -270,3 +282,188 @@ func testRollingUpdateDeploymentEvents(f *Framework) {
+			},
+			UniqueLabelKey: extensions.DefaultDeploymentUniqueLabelKey,
+			Template: api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: deploymentPodLabels,
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  "redis",
+							Image: "redis",
+						},
+					},
+				},
+			},
+		},
+	}
+	_, err = c.Deployments(ns).Create(&newDeployment)
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		Logf("deleting deployment %s", deploymentName)
+		Expect(c.Deployments(ns).Delete(deploymentName, nil)).NotTo(HaveOccurred())
+	}()
+
+	err = waitForDeploymentStatus(c, ns, deploymentName, 3, 0, 3, 0)
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func testRecreateDeploymentEvents(f *Framework) {
+	ns := f.Namespace.Name
+	c := f.Client
+	// Create nginx pods.
+	deploymentPodLabels := map[string]string{"name": "sample-pod"}
+	rcPodLabels := map[string]string{
+		"name": "sample-pod",
+		"pod":  "nginx",
+	}
+	rcName := "nginx-controller"
+	_, err := c.ReplicationControllers(ns).Create(&api.ReplicationController{
+		ObjectMeta: api.ObjectMeta{
+			Name: rcName,
+		},
+		Spec: api.ReplicationControllerSpec{
+			Replicas: 1,
+			Selector: rcPodLabels,
+			Template: &api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: rcPodLabels,
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  "nginx",
+							Image: "nginx",
+						},
+					},
+				},
+			},
+		},
+	})
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		Logf("deleting replication controller %s", rcName)
+		Expect(c.ReplicationControllers(ns).Delete(rcName)).NotTo(HaveOccurred())
+	}()
+	// Verify that the required pods have come up.
+	err = verifyPods(c, ns, "sample-pod", false, 1)
+	if err != nil {
+		Logf("error in waiting for pods to come up: %s", err)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// Create a deployment to delete nginx pods and instead bring up redis pods.
+	deploymentName := "redis-deployment"
+	Logf("Creating deployment %s", deploymentName)
+	newDeployment := extensions.Deployment{
+		ObjectMeta: api.ObjectMeta{
+			Name: deploymentName,
+		},
+		Spec: extensions.DeploymentSpec{
+			Replicas: 1,
+			Selector: deploymentPodLabels,
+			Strategy: extensions.DeploymentStrategy{
+				Type: extensions.RecreateDeploymentStrategyType,
+			},
+			UniqueLabelKey: extensions.DefaultDeploymentUniqueLabelKey,
+			Template: api.PodTemplateSpec{
+				ObjectMeta: api.ObjectMeta{
+					Labels: deploymentPodLabels,
+				},
+				Spec: api.PodSpec{
+					Containers: []api.Container{
+						{
+							Name:  "redis",
+							Image: "redis",
+						},
+					},
+				},
+			},
+		},
+	}
+	_, err = c.Deployments(ns).Create(&newDeployment)
+	Expect(err).NotTo(HaveOccurred())
+	defer func() {
+		Logf("deleting deployment %s", deploymentName)
+		Expect(c.Deployments(ns).Delete(deploymentName, nil)).NotTo(HaveOccurred())
+	}()
+
+	err = waitForDeploymentStatus(c, ns, deploymentName, 1, 0, 1, 0)
```

Comment:
By the way, it's enough to test the deployment logic to set the he replica count to 1.
For rolling update deployment with replica count 3 , should work as follows::
scale up new rc to 1
scale up new rc to 2
scale up new rc to 3
scale down old rc to 2
scale down old rc to 1
scale down old rc to 0

This logic can be verify by a deployment with replica count 1. RecreateDeployment is similiar